cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/azure_onboarder.py

```diff
@@ -49,20 +49,16 @@ def execute_action(action: TerraformAction, profiles: List[AzureProfile]) -> boo
 
 # az login is required prior to calling terraform: "get_nsg.py" uses Azure CLI to gather Network Security Group names
 def azure_login(profile: AzureProfile) -> bool:
-    azure_login_command = f"login --service-principal -u {profile.principal_id} -p {profile.principal_secret} --tenant {profile.tenant_id}"
-    return_code, _, logs = az(azure_login_command)
-    if return_code != EX_OK:
-        log.warning(
-            "Failed to sign into Azure using profile '%s' credentials. Skipping profile. Error message: '%s'",
-            profile.name,
-            logs,
-        )
+    command = f"login --service-principal -u {profile.principal_id} -p {profile.principal_secret} --tenant {profile.tenant_id}"  # returns a list
+    output_dict = az_cli(command)
+    if not isinstance(output_dict, list):
```

Comment:
I'm sorry to say that, but the `az_cli` wrapper makes the code worse. I can see that there may be a need to automatically retry Azure CLI commands (although I'm not sure about it), but why not preserve the original API. It is not beautiful, but at least it allows to easily detect success from failure without having to test type of returned data.

Reply:
az_cli will be reused in profiles tool.
we don't need the logs nor the return code from az call - they may be handled and logged in one place instead of spread  all over the place.
we need to know if the command succeeded and what data was returned. and this is what the wrapper/facade does, it adjusts the generic interface for our specific needs. IMO this is more handy than the generic code, logs, data = az(...)